cg/services/analysis_service/analysis_service.py

```diff
@@ -1,11 +1,45 @@
+from datetime import datetime
+
 from cg.apps.tb.api import TrailblazerAPI
 from cg.apps.tb.models import TrailblazerAnalysis
+from cg.constants import Workflow
+from cg.store.models import Case, Analysis
+from cg.store.store import Store
 
 
 class AnalysisService:
-    def __init__(self, analysis_client: TrailblazerAPI):
+    def __init__(self, analysis_client: TrailblazerAPI, status_db: Store):
+        self.status_db = status_db
         self.analysis_client = analysis_client
 
     def add_upload_job(self, case_id: str, slurm_id: int):
         analysis: TrailblazerAnalysis = self.analysis_client.get_latest_completed_analysis(case_id)
         self.analysis_client.add_upload_job_to_analysis(slurm_id=slurm_id, analysis_id=analysis.id)
+
+    def get_analyses_to_upload_for_workflow(self, workflow: Workflow) -> list[Analysis]:
```

Comment:
There is a function in the crud, called `cg.store.crud.read.ReadHandler.get_analyses_to_upload`. Can't we use that one? 🤔

Reply:
There is additional logic to that, it depends on both the store and trailblazer